src/proxyClient.js

```diff
@@ -97,7 +97,14 @@ var frameId = 0;
 
 // Worker
 
-var worker = new Worker('{{{ filename }}}.js');
+var filename = '{{{ filename }}}.js';
+var workerURL = filename;
+var fileBytes = tryParseAsDataURI(filename);
+if (fileBytes) {
+  workerURL = URL.createObjectURL(new Blob([fileBytes], {type: 'application/javascript'}));
+}
+
+var worker = new Worker(workerURL);
```

Comment:
There are several locations in this PR, like this, which are unconditionally applied, even though this is only relevant for `-s SINGLE_FILE=1` builds. It would be good to have the base64 machinery included only conditionally when requested, because a lot of Emscripten users complain about "hello world bloat", i.e. that the runtime is too complex for small stuff.